crates/rust-analyzer/src/bin/main.rs

```diff
@@ -119,7 +119,9 @@ fn setup_logging(log_file: Option<&Path>) -> Result<()> {
         None => None,
     };
     let filter = env::var("RA_LOG").ok();
-    logger::Logger::new(log_file, filter.as_deref()).install()?;
+    // deliberately enable all `error` logs if the user has not set RA_LOG, as there is usually useful
+    // information in there for debugging
+    logger::Logger::new(log_file, filter.as_deref().or(Some("error"))).install()?;
```

Comment:
Any arguments against this? The way this PR is structured now is that we do small error popups for build errors and others that only appear on project load and put the bulkier information into the server logs as most notification UIs (or maybe really only VSCode who knows) don't handle big messages that well.